openapi/src/apis/children_api_client.rs

```diff
@@ -0,0 +1,454 @@
+            configuration.base_path,
+            node_id = crate::apis::client::urlencode(node_id),
+            nexus_id = nexus_id.to_string(),
+            child_id = crate::apis::client::urlencode(child_id)
+        );
+        let mut local_var_req_builder =
+            local_var_client.request(awc::http::Method::DELETE, local_var_uri_str.as_str());
+
+        if let Some(ref local_var_user_agent) = configuration.user_agent {
+            local_var_req_builder = local_var_req_builder
+                .insert_header((awc::http::header::USER_AGENT, local_var_user_agent.clone()));
+        }
+        if let Some(ref local_var_token) = configuration.bearer_access_token {
+            local_var_req_builder = local_var_req_builder.bearer_auth(local_var_token.to_owned());
+        };
+        let mut local_var_resp = if configuration.trace_requests {
+            local_var_req_builder.trace_request().send().await
+        } else {
+            local_var_req_builder.send().await
+        }?;
+
+        let local_var_status = local_var_resp.status();
+
+        if local_var_status.is_success() {
+            Ok(())
+        } else {
+            match local_var_resp.json::<crate::models::RestJsonError>().await {
+                Ok(error) => Err(Error::ResponseError(ResponseContent {
+                    status: local_var_status,
+                    error,
+                })),
+                Err(_) => Err(Error::ResponseUnexpected(ResponseContentUnexpected {
+                    status: local_var_status,
+                    text: local_var_resp.json().await?,
+                })),
+            }
+        }
+    }
+    async fn get_nexus_child(
+        &self,
+        nexus_id: &str,
+        child_id: &str,
+    ) -> Result<crate::models::Child, Error<crate::models::RestJsonError>> {
+        let configuration = &self.configuration;
+        let local_var_client = &configuration.client;
+
+        let local_var_uri_str = format!(
+            "{}/nexuses/{nexus_id}/children/{child_id}",
+            configuration.base_path,
+            nexus_id = nexus_id.to_string(),
+            child_id = crate::apis::client::urlencode(child_id)
+        );
+        let mut local_var_req_builder =
+            local_var_client.request(awc::http::Method::GET, local_var_uri_str.as_str());
+
+        if let Some(ref local_var_user_agent) = configuration.user_agent {
+            local_var_req_builder = local_var_req_builder
+                .insert_header((awc::http::header::USER_AGENT, local_var_user_agent.clone()));
+        }
+        if let Some(ref local_var_token) = configuration.bearer_access_token {
+            local_var_req_builder = local_var_req_builder.bearer_auth(local_var_token.to_owned());
+        };
+        let mut local_var_resp = if configuration.trace_requests {
+            local_var_req_builder.trace_request().send().await
+        } else {
+            local_var_req_builder.send().await
+        }?;
```

Comment:
Nit: These checks get repeated a lot, we could move them to a separate function (unless this is auto generated, in which case never mind).

Reply:
Yes this is autogenerated code